packages/web/components/control/token-select-limit.tsx

```diff
@@ -0,0 +1,227 @@
+import { CoinPretty, PricePretty } from "@keplr-wallet/unit";
+import { DEFAULT_VS_CURRENCY } from "@osmosis-labs/server";
+import classNames from "classnames";
+import { observer } from "mobx-react-lite";
+import Image from "next/image";
+import { useRouter } from "next/router";
+import { FunctionComponent, useMemo, useState } from "react";
+
+import { TokenSelectDrawerLimit } from "~/components/drawers/token-select-drawer-limit";
+import { Disableable } from "~/components/types";
+import { EventName } from "~/config";
+import { useAmplitudeAnalytics, useWindowSize } from "~/hooks";
+import { OrderDirection } from "~/hooks/limit-orders";
+import { useCoinPrice } from "~/hooks/queries/assets/use-coin-price";
+import { useSwapAsset } from "~/hooks/use-swap";
+import { formatPretty } from "~/utils/formatter";
+
+export interface TokenSelectLimitProps {
+  dropdownOpen?: boolean;
+  setDropdownOpen?: (value: boolean) => void;
+  // TODO: Better typing
+  selectableAssets: ReturnType<typeof useSwapAsset>["asset"][];
```

Comment:
If you want cleaner typing perhaps you can define a `SwapAsset` type in the use-swap file that is the return type of `useSwapAsset` (or you can define it as the return value of `useSwapAsset`)

Reply:
https://github.com/osmosis-labs/osmosis-frontend/pull/3308/commits/9d8f4395e62ff823a46bdc59fb3838b80b4cdb9c